java/ingest/ingest-runner/src/test/java/sleeper/ingest/impl/IngestCoordinatorUsingDirectWriteBackedByArrowIT.java

```diff
@@ -15,131 +15,244 @@
+import sleeper.core.CommonTestConstants;
 import sleeper.core.partition.PartitionTree;
 import sleeper.core.partition.PartitionsBuilder;
 import sleeper.core.record.Record;
 import sleeper.core.schema.type.LongType;
+import sleeper.core.statestore.FileInfo;
+import sleeper.core.statestore.FileInfoFactory;
 import sleeper.ingest.impl.recordbatch.arrow.ArrowRecordBatchFactory;
 import sleeper.ingest.testutils.AwsExternalResource;
-import sleeper.ingest.testutils.IngestTestHelper;
+import sleeper.ingest.testutils.IngestCoordinatorTestParameters;
 import sleeper.ingest.testutils.RecordGenerator;
+import sleeper.ingest.testutils.ResultVerifier;
+import sleeper.ingest.testutils.TestIngestType;
+import sleeper.statestore.dynamodb.DynamoDBStateStore;
+import sleeper.statestore.dynamodb.DynamoDBStateStoreCreator;
 
 import java.nio.file.Path;
-import java.util.AbstractMap;
-import java.util.Map;
+import java.time.Instant;
+import java.util.List;
 import java.util.function.Consumer;
-import java.util.function.Function;
 import java.util.stream.Collectors;
 import java.util.stream.LongStream;
-import java.util.stream.Stream;
 
+import static java.nio.file.Files.createTempDirectory;
+import static org.assertj.core.api.Assertions.assertThat;
 import static org.assertj.core.api.Assertions.assertThatThrownBy;
+import static sleeper.configuration.testutils.LocalStackAwsV1ClientHelper.buildAwsV1Client;
+import static sleeper.ingest.testutils.ResultVerifier.readMergedRecordsFromPartitionDataFiles;
+import static sleeper.ingest.testutils.TestIngestType.directWriteBackedByArrowWriteToLocalFile;
 
+@Testcontainers
 class IngestCoordinatorUsingDirectWriteBackedByArrowIT {
-    @RegisterExtension
-    public static final AwsExternalResource AWS_EXTERNAL_RESOURCE = new AwsExternalResource(
-            LocalStackContainer.Service.S3,
-            LocalStackContainer.Service.DYNAMODB);
+    @Container
+    public static LocalStackContainer localStackContainer = new LocalStackContainer(DockerImageName.parse(CommonTestConstants.LOCALSTACK_DOCKER_IMAGE))
+            .withServices(LocalStackContainer.Service.S3, LocalStackContainer.Service.DYNAMODB);
+    private final AmazonDynamoDB dynamoDB = buildAwsV1Client(localStackContainer, LocalStackContainer.Service.DYNAMODB,
+            AmazonDynamoDBClientBuilder.standard());
+    private final Configuration configuration = AwsExternalResource.getHadoopConfiguration(localStackContainer);
+    private static final String TABLE_NAME = "test-table";
     private static final String DATA_BUCKET_NAME = "databucket";
     @TempDir
     public Path temporaryFolder;
 
-    @BeforeEach
-    public void before() {
-        AWS_EXTERNAL_RESOURCE.getS3Client().createBucket(DATA_BUCKET_NAME);
-    }
-
     @AfterEach
     public void after() {
-        AWS_EXTERNAL_RESOURCE.clear();
+        dynamoDB.deleteTable(TABLE_NAME + "-af");
+        dynamoDB.deleteTable(TABLE_NAME + "-rgcf");
+        dynamoDB.deleteTable(TABLE_NAME + "-p");
```

Comment:
We've found it unreliable and sometimes time consuming to delete DynamoDB tables after tests.

It also looks like the DynamoDB state store isn't really needed for this test class, judging by the fact that it isn't used in IngestCoordinatorUsingDirectWriteBackedByArrowRecordWriterAcceptingRecordListIT.